### PR TITLE
Convert Cadence logs to realtime

### LINKED ISSUE #302: Real-time log from Cadence
### Issue To Be Solved

Currently logs are shown after transaction or script execution. We can make them real-time, by using fvm cadence logging. 

I think after #293 , shouldn't be so hard. ( We also need for debugger )

### PR DESCRIPTION
**Pull request by bluesign**
Closes #302 

## Description

Converts cadence logs to realtime output instead of after script/transaction execution. 

______

For contributor use:

- [X] Targeted PR against `master` branch
- [X] Linked to GitHub issue with discussion and accepted design OR link to spec that describes this work
- [X] Code follows the [standards mentioned here](https://github.com/onflow/flow-emulator/blob/master/CONTRIBUTING.md#styleguides)
- [ ] Updated relevant documentation
- [X] Re-reviewed `Files changed` in the GitHub PR explorer
- [ ] Added appropriate labels

### CODE REVIEW

File: blockchain.go [hunk 1]
```diff
@@ -698,11 +727,6 @@ func (b *Blockchain) newFVMContextFromHeader(header *flowgo.Header) fvm.Context
 	return fvm.NewContextFromParent(
 		b.vmCtx,
 		fvm.WithBlockHeader(header),
-		fvm.WithReusableCadenceRuntimePool(
-			reusableRuntime.NewReusableCadenceRuntimePool(
-				1,
-				runtime.Config{Debugger: b.debugger, AccountLinkingEnabled: true}),
-		),
```
Comment (turbolent) [inline blockchain.go:705]: Was this removal intentional? Is it not needed anymore?

**Review by turbolent — COMMENTED** [review]
Nice! Great idea 👌

File: blockchain.go (hunk 1 above)
Comment (bluesign) [reply blockchain.go:705]: yeah, @m-peter pointed out[0] in coverage PR that it is not needed anymore as config will be passed via parent context.

[0] https://github.com/onflow/flow-emulator/pull/332#discussion_r1135265180

Comment (m-Peter) [reply blockchain.go:705]: Exactly, I observed that it was redundant for the `Debugger` & `AccountLinkingEnabled` config values. Moreover, the removal was necessary for the coverage reporting feature to work properly. However, at the time of writing that comment, the `ReusableCadenceRuntimePool` was responsible for also initializing the `CoverageReport` object. This was later changed. If there's no specific reason for constructing a new `ReusableCadenceRuntimePool`, then I would say that it's safe to remove.

Comment (turbolent) [reply blockchain.go:705]: Sounds good, thanks for the explanation! I was just wondering how the removal relates to this PR

**Review by turbolent — APPROVED** [review]

File: blockchain.go [hunk 2]
```diff
@@ -481,11 +494,27 @@ func (b *Blockchain) LoadSnapshot(name string) error {
...
 
+type CadenceHook struct {
+	MainLogger *zerolog.Logger
+}
+
+func (h CadenceHook) Run(e *zerolog.Event, level zerolog.Level, msg string) {
+	if level != zerolog.NoLevel && strings.HasPrefix(msg, "Cadence log:") {
+		h.MainLogger.Info().Msg(
+			strings.Replace(msg,
+				"Cadence log:",
```
Comment (turbolent) [inline blockchain.go:505]:
```suggestion
	const logPrefix = "Cadence log:"
	if level != zerolog.NoLevel && strings.HasPrefix(msg, logPrefix) {
		h.MainLogger.Info().Msg(
			strings.Replace(msg,
				logPrefix,
```

File: blockchain.go (hunk 1 above)
Comment (bluesign) [reply blockchain.go:705]: ah yeah sorry, actually it was something left from when I tested if removing was safe, then it somehow came into this PR unintentionally, but then I thought adding would not make sense, for the sake of it. Sorry created a bit confusion
